### PR TITLE
[Container] Update container cmdlets to use 2024-05-01-preview version API

### PR DESCRIPTION
**Pull request by kaushik-ms**
**Related command**
az container

**Description**

- We are in process of releasing some new features, one of them is a new azure resource called ContainerGroupProfile.
- For that, added a new command group called az container container-group-profile, this includes commands related to various resource operations like create,list,show,delete and operations related to revisions.
- Modified the az container create command to include new container group properties.
- Additionally, added the required test scenarios.

**Testing Guide**

**History Notes**

[Component Name 1] BREAKING CHANGE: `az command a`: Make some customer-facing breaking change
[Component Name 2] `az command b`: Add some customer-facing feature

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [x] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
Thank you for your contribution! We will review the pull request and get back to you soon.

File: src/azure-cli/azure/cli/command_modules/container/_help.py [hunk 1]
```diff
@@ -106,3 +106,40 @@
...
+helps['container container-group-profile'] = """
+type: group
+short-summary: Manage Azure Container Instance Container Group Profile.
+"""
+
+helps['container container-group-profile create'] = """
+type: command
+short-summary: Create a container group profile.
+examples:
+  - name: Creates a container group profile.
```
Comment (yanzhudd) [inline src/azure-cli/azure/cli/command_modules/container/_help.py:119]: please use first-person verb for all help messages.
```suggestion
  - name: Create a container group profile.
```

File: src/azure-cli/azure/cli/command_modules/container/_help.py [hunk 2]
```diff
@@ -106,3 +106,40 @@
...
+    text: az container container-group-profile create --resource-group MyResourceGroup --name myapp --image myimage:latest --cpu 1 --memory 1
+"""
+
+helps['container container-group-profile show'] = """
+type: command
+short-summary: Gets a container group profile.
```
Comment (yanzhudd) [inline src/azure-cli/azure/cli/command_modules/container/_help.py:125]:
```suggestion
short-summary: Get a container group profile.
```

File: src/azure-cli/azure/cli/command_modules/container/_help.py [hunk 3]
```diff
@@ -106,3 +106,40 @@
...
+examples:
+  - name: Creates a container group profile.
```
Comment (yanzhudd) [inline src/azure-cli/azure/cli/command_modules/container/_help.py:127]:
```suggestion
  - name: Create a container group profile.
```

File: src/azure-cli/azure/cli/command_modules/container/_help.py [hunk 4]
```diff
@@ -106,3 +106,40 @@
...
+    text: az container container-group-profile show --resource-group MyResourceGroup --name mycgprofile
+"""
+
+helps['container container-group-profile show-revision'] = """
+type: command
+short-summary: Shows a container group profile revision.
```
Comment (yanzhudd) [inline src/azure-cli/azure/cli/command_modules/container/_help.py:133]:
```suggestion
short-summary: Show a container group profile revision.
```

File: src/azure-cli/azure/cli/command_modules/container/_help.py [hunk 5]
```diff
@@ -106,3 +106,40 @@
...
+examples:
+  - name: Creates a container group profile.
```
Comment (yanzhudd) [inline src/azure-cli/azure/cli/command_modules/container/_help.py:135]:
```suggestion
  - name: Create a container group profile.
```

File: src/azure-cli/azure/cli/command_modules/container/_help.py [hunk 6]
```diff
@@ -106,3 +106,40 @@
...
+    text: az container container-group-profile show-revision --resource-group MyResourceGroup --name mycgprofile -r 1
+"""
+
+helps['container container-group-profile delete'] = """
+type: command
+short-summary: Deletes a container group profile.
+examples:
+  - name: Deletes a container group profile.
+    text: az container container-group-profile delete  --resource-group MyResourceGroup --name mycgprofile
```
Comment (yanzhudd) [inline src/azure-cli/azure/cli/command_modules/container/_help.py:144]:
```suggestion
helps['container container-group-profile delete'] = """
type: command
short-summary: Delete a container group profile.
examples:
  - name: Delete a container group profile.
    text: az container container-group-profile delete  --resource-group MyResourceGroup --name mycgprofile
```

File: src/azure-cli/azure/cli/command_modules/container/_help.py [hunk 7]
```diff
@@ -106,3 +106,40 @@
...
+    text: az container container-group-profile show --resource-group MyResourceGroup --name mycgprofile
+"""
+
+helps['container container-group-profile show-revision'] = """
+type: command
+short-summary: Shows a container group profile revision.
+examples:
+  - name: Creates a container group profile.
+    text: az container container-group-profile show-revision --resource-group MyResourceGroup --name mycgprofile -r 1
+"""
```
Comment (yanzhudd) [inline src/azure-cli/azure/cli/command_modules/container/_help.py:137]: It should be "shows a container group profile revision", right?

File: src/azure-cli/azure/cli/command_modules/container/_help.py (hunk 3 above)
Comment (yanzhudd) [inline src/azure-cli/azure/cli/command_modules/container/_help.py:127]: It should be "Get a container group profile.", right?

File: src/azure-cli/azure/cli/command_modules/container/_help.py [hunk 8]
```diff
@@ -106,3 +106,40 @@
...
+
+helps['container container-group-profile delete'] = """
+type: command
+short-summary: Deletes a container group profile.
+examples:
+  - name: Deletes a container group profile.
+    text: az container container-group-profile delete  --resource-group MyResourceGroup --name mycgprofile
+"""
```
Comment (yanzhudd) [inline src/azure-cli/azure/cli/command_modules/container/_help.py:145]:
```suggestion
helps['container container-group-profile delete'] = """
type: command
short-summary: Delete a container group profile.
examples:
  - name: Delete a container group profile.
    text: az container container-group-profile delete --resource-group MyResourceGroup --name mycgprofile
"""
```

**Comment by yanzhudd** [comment]
please note that Azure CLI is only responsible for code style and specification but not business logic, so please involve service team to review it as well.

File: src/azure-cli/azure/cli/command_modules/container/_params.py [hunk 9]
```diff
@@ -116,6 +128,14 @@ def load_arguments(self, _):
         c.argument('log_analytics_workspace', help='The Log Analytics workspace name or id. Use the current subscription or use --subscription flag to set the desired subscription.')
         c.argument('log_analytics_workspace_key', help='The Log Analytics workspace key.')
 
+    with self.argument_context('container create', arg_group='Container Group Profile Reference') as c:
+        c.argument('container_group_profile_id', help='The reference container group profile ARM resource id.')
+        c.argument('container_group_profile_revision', help='The reference container group profile revision.')
+
+    with self.argument_context('container create', arg_group='Standby Pool Profile') as c:
+        c.argument('standby_pool_profile_id', help='The standby pool profile ARM resource id from which the container will be reused.')
+        c.argument('fail_container_group_create_on_reuse_failure', help='The flag indicating whether to fail the container group creation if the standby pool reuse failed.', action='store_true')
```
Comment (yanzhudd) [inline src/azure-cli/azure/cli/command_modules/container/_params.py:137]: It is better to add a shorter option to make it more easier for users to use this param
```suggestion
        c.argument('fail_container_group_create_on_reuse_failure', help='The flag indicating whether to fail the container group creation if the standby pool reuse failed.', action='store_true')
```

File: src/azure-cli/azure/cli/command_modules/container/_help.py (hunk 7 above)
Comment (kaushik-ms) [reply src/azure-cli/azure/cli/command_modules/container/_help.py]: done

**Comment by kaushik-ms** [comment]
\azp run

**Comment by gairTanm** [comment]
Reviewed, LGTM

**Review by jrathore-msft — APPROVED** [review]

**Review by yanzhudd — APPROVED** [review]

**Review by zhoxing-ms — APPROVED** [review]

**Comment by zhoxing-ms** [comment]
@kaushik-ms Please refer to doc https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests to add description information for new parameters in the PR description

**Pull request merged**